packages/account/src/Sections/Security/ConnectedApps/__tests__/connected-apps.spec.js

```diff
@@ -0,0 +1,77 @@
+import React from 'react';
+import { render, screen, waitFor, waitForElementToBeRemoved, fireEvent } from '@testing-library/react';
+import ConnectedApps from '../connected-apps';
+import { act } from '@testing-library/react';
+
+const oauth_apps_list = [
+    {
+        name: 'Local',
+        app_markup_percentage: 0,
+        app_id: 9999,
+        scopes: ['read', 'admin', 'trade', 'payments'],
+        last_used: '2021-10-31 06:49:52',
+    },
+];
+
+jest.mock('@deriv/shared/src/services/ws-methods', () => ({
+    __esModule: true,
+    default: 'mockedDefaultExport',
+    WS: {
+        authorized: {
+            send: ({ oauth_apps, revoke_oauth_app }) => {
+                if (revoke_oauth_app) oauth_apps_list.pop();
+                return { oauth_apps: oauth_apps_list };
+            },
+        },
+    },
+}));
+
+const modalRoot = document.createElement('div');
+modalRoot.setAttribute('id', 'modal_root');
+document.body.appendChild(modalRoot);
+
+describe('Connected Apps', () => {
+    const originalOffsetHeight = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetHeight');
+    const originalOffsetWidth = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'offsetWidth');
+
+    beforeAll(() => {
+        Object.defineProperty(HTMLElement.prototype, 'offsetHeight', { configurable: true, value: 50 });
+        Object.defineProperty(HTMLElement.prototype, 'offsetWidth', { configurable: true, value: 50 });
+    });
+
+    afterAll(() => {
+        Object.defineProperty(HTMLElement.prototype, 'offsetHeight', originalOffsetHeight);
+        Object.defineProperty(HTMLElement.prototype, 'offsetWidth', originalOffsetWidth);
+    });
+
+    test('renders correctly', async () => {
+        const { container } = render(<ConnectedApps />);
+
+        expect(screen.getByText(/Authorised applications/i)).toBeInTheDocument();
+
+        await waitForElementToBeRemoved(() => container.querySelector('.initial-loader'));
+
+        await waitFor(() => {
+            expect(screen.getByText('Local')).toBeInTheDocument();
+            expect(screen.getByText(oauth_apps_list[0].last_used)).toBeInTheDocument();
+            expect(screen.getByText('Revoke access')).toBeInTheDocument();
+        });
+    });
+
+    test('revoke access when click on confirm', async () => {
+        const { container } = render(<ConnectedApps />);
+
+        await waitForElementToBeRemoved(() => container.querySelector('.initial-loader'));
+        fireEvent.click(screen.getByText('Revoke access'));
+
+        await waitFor(() => {
+            expect(screen.getByText('Confirm')).toBeInTheDocument();
+            expect(screen.getByText('Back')).toBeInTheDocument();
+        });
+
+        act(() => {
+            fireEvent.click(screen.getByText('Confirm'));
+        });
```

Comment:
(question) `fireEvent` is already wrapped in `act`. possible to remove this?

Reply:
@hojjat-fs @reza-fs I'm wondering why we need act here. This part is not clear to me.  We also have at disposal user-event in testing-library which itself uses fireEvent inside  https://testing-library.com/docs/ecosystem-user-event/ Maybe it would be useful and more descriptive way? Can you check?

Reply:
We need to wrap the code rendering it and performing updates inside an act() call to make tests run closer to how React works in the browser

https://reactjs.org/docs/test-utils.html#act

https://github.com/threepointone/react-act-examples/blob/master/sync.md